include/zephyr/drivers/video.h

```diff
@@ -833,6 +833,17 @@ void video_closest_frmival(const struct device *dev, enum video_endpoint_id ep,
  */
 #define VIDEO_FOURCC_FROM_STR(str) VIDEO_FOURCC((str)[0], (str)[1], (str)[2], (str)[3])
 
+/**
+ * @brief Convert a four-character-code to a four-character-string
+ *
+ * Convert a string literal or variable into a four-character-string
+ * as defined by @ref VIDEO_FOURCC.
+ *
+ * @param fcc 32-bit integer to be converted
+ * @return Four-character-string built out of this fourcc.
+ */
+#define VIDEO_FOURCC_TO_STR(u) ((char []){(u) >> 0, (u) >> 8, (u) >> 16, (u) >> 24, '\0'})
```

Comment:
don't each byte need to explicitly be cast to char? probably warning-prone otherwise?
also not sure about the naming `u`, and I think you might face endianness issues?

```c
#define VIDEO_FOURCC_TO_STR(fourcc) ((char []){ \
    (char)((fourcc) & 0xFF), \
    (char)(((fourcc) >> 8) & 0xFF), \
    (char)(((fourcc) >> 16) & 0xFF), \
    (char)(((fourcc) >> 24) & 0xFF), \
    '\0' \
})
```

Reply:
> you might face endianness issues

The `pixelformat` used everywhere in `video.h` is what FOURCC are used for, and is always in the native endianess of the CPU. That can lead to surprises if logging something directly from the network, I added a comment.

Thank you for the improvement and reminding that clarity and correctness are better than compactness.

Reply:
- little/big-endian (LSB/MSB first)
- native-endian (what the CPU has)
- confused-endian (the opposite) [like FOURCCs](https://en.wikipedia.org/wiki/FourCC#Compiler_support)